src/main/java/com/epam/aidial/core/security/RuleMatcher.java

```diff
@@ -0,0 +1,81 @@
+package com.epam.aidial.core.security;
+
+import com.epam.aidial.core.ProxyContext;
+import com.epam.aidial.core.data.Rule;
+import lombok.experimental.UtilityClass;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@UtilityClass
+public class RuleMatcher {
```

Comment:
can you write some unit tests to cover this class?

Reply:
added